app/controllers/topics_controller.rb

```diff
@@ -821,7 +821,7 @@ def convert_topic
     guardian.ensure_can_convert_topic!(topic)
 
     if params[:type] == "public"
-      converted_topic = topic.convert_to_public_topic(current_user)
+      converted_topic = topic.convert_to_public_topic(current_user, params[:category_id])
```

Comment:
Are we ensuring the current user is able to post in that category somewhere down the line?

Reply:
No but only staff members can convert topics from PMs to public, and staff members can [change any category](https://github.com/discourse/discourse/blob/master/lib/guardian/topic_guardian.rb#L57).